[BUG] Linux clang build fails due to -Werror,-Wshorten-64-to-32

### Reproduction steps

```bash
Compile example apps for Linux with clang


Step #5 - "Linux": 2022-10-17 16:59:57 INFO    clang++ -MMD -MF obj/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/pw_protobuf.decoder.cc.o.d --target=aarch64-linux-gnu -O0 -g2 -fno-common -ffunction-sections -fdata-sections -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -Wall -Werror -Wextra -Wshadow -Wunreachable-code -Wvla -Wformat -Wformat-nonliteral -Wformat-security -Wimplicit-fallthrough -Wheader-hygiene -Wshorten-64-to-32 -Wformat-type-confusion -Wno-deprecated-declarations -Wno-unknown-warning-option -Wno-maybe-uninitialized -Wno-missing-field-initializers -Wno-unused-parameter -fdiagnostics-color --sysroot=/opt/ubuntu-21.04-aarch64-sysroot -fno-strict-aliasing -fdiagnostics-color -g -fno-common -fno-exceptions -ffunction-sections -fdata-sections -Wall -Wextra -Wimplicit-fallthrough -Wcast-qual -Wundef -Wpointer-arith -Werror -Wno-error=cpp -Wno-error=deprecated-declarations -ffile-prefix-map=/workspace/out/linux-arm64-light-rpc-ipv6only-clang=out -ffile-prefix-map=/workspace/examples/lighting-app/linux/= -ffile-prefix-map=../../examples/lighting-app/linux/= -ffile-prefix-map=/workspace/out/linux-arm64-light-rpc-ipv6only-clang=out -std=gnu++17 -fno-rtti -Wnon-virtual-dtor -fno-rtti -Wnon-virtual-dtor -std=c++17 -Wno-register  -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_bytes/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_containers/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_assert/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_assert/assert_compatibility_public_overrides -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_preprocessor/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_assert_log/check_backend_public_overrides -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_assert_log/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_log/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_log_basic/public_overrides -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_log_basic/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_stream/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_polyfill/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_polyfill/standard_library_public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_span/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_status/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_result/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_varint/public -I../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_function/public -c ../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/decoder.cc -o obj/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/pw_protobuf.decoder.cc.o
Step #5 - "Linux": 2022-10-17 16:59:57 INFO    ../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/decoder.cc:56:19: error: implicit conversion loses integer precision: 'uint64_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
Step #5 - "Linux": 2022-10-17 16:59:57 INFO      return FieldKey(key).field_number();
Step #5 - "Linux": 2022-10-17 16:59:57 INFO             ~~~~~~~~ ^~~
Step #5 - "Linux": 2022-10-17 16:59:57 INFO    ../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/decoder.cc:68:10: error: implicit conversion loses integer precision: 'uint64_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
Step #5 - "Linux": 2022-10-17 16:59:57 INFO      *out = value;
Step #5 - "Linux": 2022-10-17 16:59:57 INFO           ~ ^~~~~
Step #5 - "Linux": 2022-10-17 16:59:57 INFO    ../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/decoder.cc:81:10: error: implicit conversion loses integer precision: 'int64_t' (aka 'long') to 'int32_t' (aka 'int') [-Werror,-Wshorten-64-to-32]
Step #5 - "Linux": 2022-10-17 16:59:57 INFO      *out = value;
Step #5 - "Linux": 2022-10-17 16:59:57 INFO           ~ ^~~~~
Step #5 - "Linux": 2022-10-17 16:59:57 INFO    ../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/decoder.cc:127:20: error: implicit conversion loses integer precision: 'uint64_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
Step #5 - "Linux": 2022-10-17 16:59:57 INFO      switch (FieldKey(key).wire_type()) {
Step #5 - "Linux": 2022-10-17 16:59:57 INFO              ~~~~~~~~ ^~~
Step #5 - "Linux": 2022-10-17 16:59:57 INFO    ../../examples/lighting-app/linux/third_party/connectedhomeip/third_party/pigweed/repo/pw_protobuf/decoder.cc:171:16: error: implicit conversion loses integer precision: 'uint64_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
Step #5 - "Linux": 2022-10-17 16:59:57 INFO      if (FieldKey(key).wire_type() != expected_type) {
Step #5 - "Linux": 2022-10-17 16:59:57 INFO          ~~~~~~~~ ^~~
Step #5 - "Linux": 2022-10-17 16:59:57 INFO    5 errors generated.
```
```


### Bug prevalence

Every time building example apps with clang

### GitHub hash of the SDK that was being used

d520b39a4cfa9d4b0ed1845adccdb87216dd0e19

### Platform

other

### Platform Version(s)

_No response_

### Anything else?

Issue started when the warning was enabled in https://github.com/project-chip/connectedhomeip/commit/d520b39a4cfa9d4b0ed1845adccdb87216dd0e19 . Looks like pigweed / example apps need to be updated as well.

--- a/.github/workflows/examples-linux-arm.yaml
+++ b/.github/workflows/examples-linux-arm.yaml
@@ -73,6 +73,7 @@ jobs:
                         --target linux-arm64-chip-tool-ipv6only-clang \
                         --target linux-arm64-lock-clang \
                         --target linux-arm64-minmdns-clang \
+                        --target linux-arm64-light-rpc-ipv6only-clang \
                         --target linux-arm64-thermostat-no-ble-clang \
                         build \
                      "

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -259,6 +259,14 @@ config("strict_warnings") {
       "-Wshorten-64-to-32",
       "-Wformat-type-confusion",
     ]
+
+    # TODO: can make this back fatal in once pigweed updates can be take again
+    #
+    # Currently `./scripts/build/build_examples.py --target linux-arm64-light-rpc-ipv6only-clang build`
+    # fails in third_party/pigweed/repo/pw_protobuf 
+    if ((current_cpu == "arm64") && (current_os == "linux")) {
+      cflags += [ "-Wno-error=shorten-64-to-32" ]
+    }
   }
 
   if (!is_asan && (current_os == "linux" || current_os == "android")) {

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -263,8 +263,8 @@ config("strict_warnings") {
     # TODO: can make this back fatal in once pigweed updates can be take again
     #
     # Currently `./scripts/build/build_examples.py --target linux-arm64-light-rpc-ipv6only-clang build`
-    # fails in third_party/pigweed/repo/pw_protobuf 
-    if ((current_cpu == "arm64") && (current_os == "linux")) {
+    # fails in third_party/pigweed/repo/pw_protobuf
+    if (current_cpu == "arm64" && current_os == "linux") {
       cflags += [ "-Wno-error=shorten-64-to-32" ]
     }
   }

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -260,7 +260,8 @@ config("strict_warnings") {
       "-Wformat-type-confusion",
     ]
 
-    # TODO: can make this back fatal in once pigweed updates can be take again
+    # TODO: can make this back fatal in once pigweed updates can be taken again.
+    #       See https://github.com/project-chip/connectedhomeip/pull/22079
     #
     # Currently `./scripts/build/build_examples.py --target linux-arm64-light-rpc-ipv6only-clang build`
     # fails in third_party/pigweed/repo/pw_protobuf

